### PR TITLE
Introduction to Agile Development

### PR DESCRIPTION
**Pull request by joanneong**
This chapter provides a simple introduction to agile development.

please help me to take a look at this :)

### CODE REVIEW

File: contents/projectManagement/AgileDevelopment.md
Comment (tran-tien-dat) [inline contents/projectManagement/AgileDevelopment.md]: This sentence is automatically attached to point number 4 above. Is it intended?

File: contents/projectManagement/AgileDevelopment.md
Comment (tran-tien-dat) [inline contents/projectManagement/AgileDevelopment.md]: This sentence structure `Agile development is essentially a software development method that ...` suggests that A is a concrete method like scrum. However, I think agile development is more of an approach to software development as explained by [wikipedia](https://en.wikipedia.org/wiki/Agile_software_development), and from this approach people derive actual concrete development framework. So I think you should change this sentence to reflect that.

File: contents/projectManagement/AgileDevelopment.md
Comment (tran-tien-dat) [inline contents/projectManagement/AgileDevelopment.md]: I still don't get what this scenario is supposed to illustrate. I think you should add some commentary to explain more.

**Review by tran-tien-dat — COMMENTED** [review]
Nice introduction to agile development. Given that the intended audience is SE students, I think you can focus less on the abstract high level things like `Structural rigidity in larger organisations` or `Unfamiliarity with agile development`. I would prefer it if you provide more concrete details on what people do in an agile development process, i.e. those things like the daily stand-up meeting you mentioned.

File: contents/projectManagement/AgileDevelopment.md
Comment (leeyh20) [inline contents/projectManagement/AgileDevelopment.md]: This example seems to illustrate why we should not take a "general plan" and apply it to a novel set of circumstances but does not show how planning is done continuously throughout a project. Maybe it will be better to give an example of a project where the requirements had to change?

File: contents/projectManagement/AgileDevelopment.md
Comment (leeyh20) [inline contents/projectManagement/AgileDevelopment.md]: Maybe it'll be good to mention the different frameworks here that have been "born" from agile development like scrum, not at the end? 

Would it be useful to mention what the manifesto pushes for? The manifesto is famous for the 12 principles of agile development, does that "define" agile?

File: contents/projectManagement/AgileDevelopment.md
Comment (leeyh20) [inline contents/projectManagement/AgileDevelopment.md]: Might be good to use H3 or H4 titles here like ### Flexible, ### Iterative so that it is more readable

File: contents/projectManagement/AgileDevelopment.md
Comment (leeyh20) [inline contents/projectManagement/AgileDevelopment.md]: Since agile has different frameworks under it, the "team roles" does it apply broadly to all the frameworks? The link seems to suggest it is for scrum.

File: contents/projectManagement/AgileDevelopment.md
Comment (leeyh20) [inline contents/projectManagement/AgileDevelopment.md:46]: Might be good to have interesting examples here by real companies, if any.

**Review by leeyh20 — CHANGES_REQUESTED** [review]
Nice read, love the variety of extra resources and links at the end. I understand that it's actually pretty long already, especially the pros and cons, but I would have liked more examples or a case study too!

File: contents/projectManagement/AgileDevelopment.md
Comment (joanneong) [reply contents/projectManagement/AgileDevelopment.md:46]: I have decided to create a new subsection with a full example at the conclusion, so hopefully it will serve the purpose here as well :)

**Comment by joanneong** [comment]
@tran-tien-dat @leeyh20 Thank you both for the feedback. They were helpful, and I really appreciate them :) Do let me know if the way it has turned out now is still lacking!

**Review by tran-tien-dat — COMMENTED** [review]
I think the Spotify example really adds stronger support for the points in the book chapter. Good job 😄 !

**Review by leeyh20 — APPROVED** [review]

**Comment by leeyh20** [comment]
Looks great!

File: contents/projectManagement/AgileDevelopment.md
Comment (leeyh20) [reply contents/projectManagement/AgileDevelopment.md:46]: :D

**Comment by joanneong** [comment]
@damithc Please take a look at this, and let me know if I can further improve it!

File: contents/projectManagement/AgileDevelopment.md
Comment (damithc) [inline contents/projectManagement/AgileDevelopment.md]: Indent this so that it appears as part of the numbered item ?

File: contents/projectManagement/AgileDevelopment.md
Comment (damithc) [inline contents/projectManagement/AgileDevelopment.md]: Indent under the bullet point?
>* **Unfamiliarity with agile development**
>  If the team is unfamiliar with agile techniques, you will not be able to reap the full benefits of agile development since agile development depends on tight collaboration between team members. Therefore, it is crucial to ensure that the entire team has good foundations in agile techniques before the transition to agile development. This can be achieved by investing early in training sessions for the team, and bring everyone in the team up-to-date with what agile development entails for the project

File: contents/projectManagement/AgileDevelopment.md
Comment (damithc) [inline contents/projectManagement/AgileDevelopment.md:31]: If you took this image from elsewhere, cite source.

**Review by damithc — CHANGES_REQUESTED** [review]
Good work. Just a few minor comments.

File: contents/projectManagement/AgileDevelopment.md
Comment (joanneong) [reply contents/projectManagement/AgileDevelopment.md:31]: I created this image using Microsoft Powerpoint :P

**Comment by joanneong** [comment]
@damithc Ready for review!

**Review by damithc — APPROVED** [review]
LGTM :+1:  if you were to add more to this in a future PR, you can address criticisms of the agile approach.

**Review by tran-tien-dat — APPROVED** [review]

**Comment by joanneong** [comment]
@damithc Am I expected to address criticisms of the agile approach for this pull request before it can be merged?

**Comment by damithc** [comment]
Not required. Can go ahead and merge. Do a squash-merge.

**Comment by joanneong** [comment]
@damithc Does that imply that I am supposed to have write access to this repository? At the moment I do not have the permission to do so, which is why I have left this here for a while...

**Comment by damithc** [comment]
> @damithc Does that imply that I am supposed to have write access to this repository? At the moment I do not have the permission to do so, which is why I have left this here for a while...

Merged :-)

**Pull request merged**
